flytepropeller/pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -70,8 +68,14 @@ func (r *ResourceLevelMonitor) countList(ctx context.Context, objects []interfac
 func (r *ResourceLevelMonitor) collect(ctx context.Context) {
 	// Emit gauges at the namespace layer - since these are just K8s resources, we cannot be guaranteed to have the necessary
 	// information to derive project/domain
-	objects := r.sharedInformer.GetStore().List()
-	counts := r.countList(ctx, objects)
+
+	pods := &corev1.PodList{}
+	if err := r.kubeClient.GetClient().List(ctx, pods); err != nil {
```

Comment:
I am uncertain about the change here. Originally, sharedInformer fetches the resources it listens to, but now it changes to only fetches pods. Also, the code seems be metrics collecting and is from four years ago. I am not sure if the code is still relevant. cc @wild-endeavor

Reply:
Agreed, it is not safe to update this to a `PodList` because it will not capture other k8s resource types for other Flyte plugins (ex. Spark, Ray, TF, PyTorch, etc).